packages/legend-query/src/stores/QueryBuilderLambdaProcessor.ts

```diff
@@ -452,31 +443,54 @@ export class QueryBuilderLambdaProcessor
         new QueryBuilderLambdaProcessor(this.queryBuilderState, undefined),
       );
 
-      // check caller
-      assertTrue(
+      if (
         matchFunctionName(
           precedingExpression.functionName,
           SUPPORTED_FUNCTIONS.GET_ALL,
-        ),
-        `Can't process filter() expression: only support filter() immediately following getAll()`,
-      );
-
-      const filterLambda = valueSpecification.parametersValues[1];
-      assertType(
-        filterLambda,
-        LambdaFunctionInstanceValue,
-        `Can't process filter() expression: filter() expects argument #1 to be a lambda function`,
-      );
-      processFilterLambda(filterLambda, filterState);
-      /**
-       * NOTE: Since group operations like and/or do not take more than 2 parameters, if there are
-       * more than 2 clauses in each group operations, then these clauses are converted into an
-       * unbalanced tree. However, this would look quite bad for UX, as such, we simplify the tree.
-       * After building the filter state.
-       */
-      filterState.simplifyTree();
+        )
+      ) {
+        const filterLambda = valueSpecification.parametersValues[1];
```

Comment:
Let's add an assertion here so if people specify fully `SUPPORTED_FUNCTIONS.TDS_FILTER` we will fail.